contributed_definitions/nyaml/NXcalibration.yaml

```diff
@@ -49,13 +60,38 @@ NXcalibration(NXobject):
     unit: NX_ANY
     doc: |
       A vector representing the axis after calibration, matching the data length
+    \@output_path:
+      doc: |
+        The path to which this data is written, e.g., the calibrated energy.
+        Should be a valid NeXus path name, e.g., /entry/data/energy.
     dimensions:
       rank: 1
       dim: [[1, ncal]]
   original_axis(NX_FLOAT):
     unit: NX_ANY
     doc: |
       Vector containing the data coordinates in the original uncalibrated axis
+    \@symbol:
+      doc: |
+        The symbol of the axis to be used in the fit_function, e.g., `energy`, `E`.
+        This should comply to python variable naming conventions.
+    \@input_path:
+      doc: |
+        The path from which this data is derived, e.g., raw detector axis.
+        Should be a valid NeXus path name, e.g., /entry/instrument/detector/raw.
+    dimensions:
+      rank: 1
+      dim: [[1, ncal]]
+  input_SYMBOL(NX_FLOAT):
```

Comment:
In #52, we additionally have
```yaml
(NXcalibration)
  ...
  calibration_file(NXnote):
    doc: |
      File containing the input data for calibration.
  MAPPING(NXdata):
    doc: |
      Mapping data for calibration.
      
      This can be used to map data points from uncalibrated
      to calibrated values, e.g., by multiplying each point by the
      corresponding point in the mapping data.
```
I guess mapping will be replaced by input_SYMBOL, correct? Do we want to keep calibration_file

Reply:
True, but we could keep `MAPPING` as a convenient short notation similar to the `scaling`/`offset` parameters we have for linear calibration in cases we don't want to specify a formula but just want to write a mapping.

For the calibration file: Wouldn't it be better if we copied the data inside the nexus file to have it directly available? If we just have a file reference that can easily get lost. If we want to copy the full file contents for ASCII file here, then it is the question what happens with binary calibration files...

Reply:
For the calibration file, I was thinking more about propietary formats for which the contents of the calibration file often cannot be read out. For example, our calibration files are typically closed, but I would still like to know which file I used. It is not really FAIR, but is useful for the user to know. This is why I had it originally as `  calibration_file(NX_CHAR)`.

Reply:
Ok, let's keep `MAPPING`, but I suggest that we _only_ use it for multiplying each point by the corresponding point in the mapping data:
```yaml
  MAPPING(NXdata):
    doc: |
      Mapping data for calibration.
      
      This can be used to map data points from uncalibrated to calibrated values,
      i.e., by multiplying each point in the input axis by the corresponding point in the mapping data.
```
If one wants to more with such input data, it is anyway needed to use `input_SYMBOL` and `fit_function`.

Reply:
Yes, lets do it like this 👍️

Reply:
Did you add `MAPPING` here @domna? I may be missing it, but I can't find it. If not, would you mind adding it?

Reply:
Forgot it. Will add it. Thanks